core/src/main/java/com/amplifyframework/datastore/DataStoreCategoryBehavior.java

```diff
@@ -48,6 +48,21 @@
             @NonNull T item,
             @NonNull ResultListener<DataStoreItemChange<T>> saveItemListener);
 
+    /**
+     * Saves an item into the DataStore if the data being overwritten meets
+     * the provided conditions. This is useful for making sure that no data
+     * is overwritten with an outdated/incorrect assumption.
```

Comment:
It applies the predicate to existing data in database, rather than the data being saved. So predicate wouldn't do anything for creation is what my understanding is.